padertorch/data/fragmenter.py

```diff
@@ -15,110 +16,199 @@
+
+        # Discard examples that are shorter than `length`
+        if to_segment_length < self.length:
+            import lazy_dataset
+            raise lazy_dataset.FilterException()
+
+        boundaries, segmented = self.segment(to_segment, to_segment_length,
+                                             axis=axis)
+
+        segmented_examples = list()
+        if self.copy_keys is not None:
+            example = {key: example.pop(key) for key in self.copy_keys}
+        for idx, (start, stop) in enumerate(boundaries):
+            example_copy = copy(example)
+            example_copy.update({to_segment_keys[i]: value[idx]
+                                 for i, value in enumerate(segmented)})
+
+            example_copy.update(segment_start=start, segment_stop=stop)
+            segmented_examples.append(deflatten(example_copy))
+        return segmented_examples
+
+
+    def segment(self, to_segment, to_segment_length, axis):
+
+        anchor = get_anchor(
+            to_segment_length, self.length, self.shift,
+            mode=self.anchor_mode, rng=self.rng
+        )
+
+        boundaries = get_segment_boundaries(
+            to_segment_length, self.length, self.shift,
+            anchor=self.anchor_mode, rng=self.rng
+        )
+
+        segmented = [segment(signal, length=self.length, shift=self.shift,
+                       rng=self.rng, axis=axis[i], anchor=anchor)
+                     for i, signal in enumerate(to_segment)]
+        return boundaries, segmented
+
+    def get_to_segment_keys(self, example):
+        if self.include is None:
+            return [
+                key for key, value in example.items()
+                if not key in self.exclude and isinstance(value, np.ndarray)
+            ]
+        else:
+            return [
+                key for key in example.keys()
+                if not key in self.exclude and
+                any([include_key in key for include_key in self.include])
+            ]
+
+    def get_axis_list(self, to_segment_keys):
+        if isinstance(self.axis, int):
+            return [self.axis] * len(to_segment_keys)
+        elif isinstance(self.axis, dict):
+            assert all([key in self.axis.keys() for key in to_segment_keys]), (
+                f'The dictionary for axis did not include keys for all'
+                f'segment arrays. axis keys: {self.axis.keys()},'
+                f'segment array keys {to_segment_keys}'
+            )
+            return [self.axis[key] for key in to_segment_keys]
+        elif isinstance(self.axis, list):
+            assert len(self.axis) == len(to_segment_keys), (
+                f'The list for axis does not include a axis for each'
+                f'segment array. axis: {self.axis}, '
+                f'segment array keys {to_segment_keys}'
+            )
+            return self.axis
+        else:
+            raise NotImplementedError('This should never be reached')
```

Comment:
`TypeError`?

Reply:
Add the type and value of axis to the error msg

Reply:
I have not done that because this error should already be cached in the init. Somebody has to overwrite self.axis for this to happen.

Reply:
Ah, ok. Nevertheless, please add axis, just in case the code gets modified: `NotImplementedError('This should never be reached', self.axis)`